### PR TITLE
Add files validation to dxFileUploader(maxSize, minSize and allowExtensions)

### PR DESCRIPTION
**Pull request by andrewmakarov**

### CODE REVIEW

File: js/localization/messages/ru.json [hunk 1]
```diff
@@ -55,6 +55,10 @@
         "dxFileUploader-uploaded": "Загружено",
         "dxFileUploader-readyToUpload": "Готово к загрузке",
         "dxFileUploader-uploadFailedMessage": "Загрузка не удалась",
+        "dxFileUploader-invalidFileExtension": "",
+        "dxFileUploader-invalidMaxFileSize": "",
+        "dxFileUploader-invalidMinFileSize": "",
+
```
Comment (jsdmitry) [inline js/localization/messages/ru.json:61]: Add messages on the Russian language.

File: js/ui/file_uploader.js [hunk 2]
```diff
@@ -273,6 +275,57 @@ var FileUploader = Editor.inherit({
             */
             onUploadAborted: null,
 
+
+            /**
+            * @name dxFileUploaderOptions.validation
+            * @type Object
+            * @default { allowedFileExtensions: [], maxFileSize: 0, minFileSize: 0, invalidFileExtensionMessage: "", invalidMaxFileSizeMessage: "", invalidMinFileSizeMessage: ""}
```
Comment (jsdmitry) [inline js/ui/file_uploader.js:282]: Add `* @default undefined` instead of this line.

File: js/ui/file_uploader.js [hunk 3]
```diff
@@ -521,9 +574,41 @@ var FileUploader = Editor.inherit({
...
+                return true;
+            }
+        }
+        return false;
+    },
+    _validateMaxFileSize: function(file) {
+        var fileSize = file.value.size;
+        var maxFileSize = this.option("validation.maxFileSize");
+        maxFileSize = maxFileSize === 0 ? fileSize : maxFileSize;
+        return fileSize <= maxFileSize;
```
Comment (jsdmitry) [inline js/ui/file_uploader.js:604]: May be `return  maxFileSize > 0 ? fileSize <= maxFileSize : true;`

File: js/ui/file_uploader.js [hunk 4]
```diff
@@ -521,9 +574,41 @@ var FileUploader = Editor.inherit({
...
+    },
+    _validateMinFileSize: function(file) {
+        var fileSize = file.value.size;
+        var minFileSize = this.option("validation.minFileSize");
+        minFileSize = minFileSize === 0 ? fileSize : minFileSize;
+        return fileSize >= minFileSize;
```
Comment (jsdmitry) [inline js/ui/file_uploader.js:610]: And `return minFileSize > 0 ? fileSize >= minFileSize : true;`

File: testing/tests/DevExpress.ui.widgets.editors/fileUploader.tests.js [hunk 5]
```diff
@@ -65,6 +67,36 @@ var fakeFile1 = {
...
+    size: 4000,
+    type: "image/jpeg",
+    lastModifiedDate: $.now()
+};
+var fakeFile4 = {
+    name: "image.gif",
+    size: 4000,
+    type: "image/gif",
+    lastModifiedDate: $.now()
+};
```
Comment (jsdmitry) [inline testing/tests/DevExpress.ui.widgets.editors/fileUploader.tests.js:87]: Why many fake files, I think three will suffice?

**Review by jsdmitry — CHANGES_REQUESTED** [review]
Add the ru messages and make refactoring

File: js/ui/file_uploader.js (hunk 2 above)
Comment (andrewmakarov) [reply js/ui/file_uploader.js:282]: In the menu widjet, I found a similar example with a complex property, where the default value is an object with all properties(dxMenuOptions.showFirstSubmenuMode)

File: js/ui/file_uploader.js (hunk 3 above)
Comment (andrewmakarov) [reply js/ui/file_uploader.js]: The ternary operator is a condition and depends on the condition, it returns true or false. You proposed a ternary operator, which in the case of true, perform one more condition.
It seems to me that in this case the readability of the code will be worse

File: js/ui/file_uploader.js (hunk 2 above)
Comment (RomanTsukanov) [reply js/ui/file_uploader.js:282]: Just remove the default from here. We should do the same to the showFirstSubmenuMode and many other options that accept objects. Defaults in them [look ricidulous](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxPopup/Configuration/animation/).

File: js/localization/messages/en.json [hunk 6]
```diff
@@ -62,6 +62,9 @@
         "dxFileUploader-uploaded": "Uploaded",
         "dxFileUploader-readyToUpload": "Ready to upload",
         "dxFileUploader-uploadFailedMessage": "Upload failed",
+        "dxFileUploader-invalidFileExtension": "File type not allowed",
+        "dxFileUploader-invalidMaxFileSize": "File size too large",
+        "dxFileUploader-invalidMinFileSize": "File size too small",
```
Comment (RomanTsukanov) [inline js/localization/messages/en.json:67]: Please change the default messages to these:

```
"dxFileUploader-invalidFileExtension": "File type is not allowed",
"dxFileUploader-invalidMaxFileSize": "File is too large",
"dxFileUploader-invalidMinFileSize": "File is too small"
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**